tests/gql_behave/tests/memgraph_V1/features/memgraph_allshortest.feature

```diff
@@ -1,191 +1,205 @@
+        And having executed:
+            """
+            CREATE (n {a:'0'})-[:r {w: DURATION({seconds: -1})}]->({a:'1'})-[:r {w: DURATION('PT2S')}]->({a:'2'}), (n)-[:r {w: DURATION('PT4S')}]->({a:'3'})
+            """
+        When executing query:
+            """
+            MATCH (n {a:'0'})-[le *allShortest 10 (e, n | e.w ) w]->(m) RETURN m.a, size(le) as s, w
+            """
+        Then an error should be raised
 
-  Scenario: Test match allShortest weight mixed numeric and duration as weights
-      Given an empty graph
-      And having executed:
-          """
-          CREATE (n {a:'0'})-[:r {w: 2}]->({a:'1'})-[:r {w: DURATION('PT2S')}]->({a:'2'}), (n)-[:r {w: DURATION('PT4S')}]->({a:'3'})
-          """
-      When executing query:
-          """
-          MATCH (n {a:'0'})-[le *allShortest 10 (e, n | e.w ) w]->(m) RETURN m.a, size(le) as s, w
-          """
-      Then an error should be raised
+    Scenario: Test match allShortest weight mixed numeric and duration as weights
+        Given an empty graph
+        And having executed:
+            """
+            CREATE (n {a:'0'})-[:r {w: 2}]->({a:'1'})-[:r {w: DURATION('PT2S')}]->({a:'2'}), (n)-[:r {w: DURATION('PT4S')}]->({a:'3'})
+            """
+        When executing query:
+            """
+            MATCH (n {a:'0'})-[le *allShortest 10 (e, n | e.w ) w]->(m) RETURN m.a, size(le) as s, w
+            """
+        Then an error should be raised
 
-  Scenario: Test allShortest return both paths of same length
-      Given an empty graph
-      And having executed:
-          """
-          CREATE (n {a:'0'})-[:r {w: 2}]->({a:'1'})-[:r {w: 3}]->({a:'2'}), (n)-[:r {w: 5}]->({a:'2'})
-          """
-      When executing query:
-          """
-          MATCH path=(n {a:'0'})-[r *allShortest (e, n | e.w ) w]->(m {a:'2'}) RETURN COUNT(path);
-          """
-      Then the result should be:
-          | COUNT(path) |
-          | 2 |
+    Scenario: Test allShortest return both paths of same length
+        Given an empty graph
+        And having executed:
+            """
+            CREATE (n {a:'0'})-[:r {w: 2}]->({a:'1'})-[:r {w: 3}]->({a:'2'}), (n)-[:r {w: 5}]->({a:'2'})
+            """
+        When executing query:
+            """
+            MATCH path=(n {a:'0'})-[r *allShortest (e, n | e.w ) w]->(m {a:'2'}) RETURN COUNT(path);
+            """
+        Then the result should be:
+            | COUNT(path) |
+            | 2 |
+
+    Scenario: Test allShortest on different edge between two nodes
+        Given an empty graph
+        And having executed:
+            """
+            CREATE (n:One), (o:Two), (m:Three), (n)-[:TYPE {cost: 0.3}]->(o), (n)-[:TYPE {cost: 40}]->(m), (n)-[:TYPE {cost: 20}]->(m)
+            """
+        When executing query:
+            """
+            MATCH p=(h:One)-[r*allshortest ..5 (e, v | e.cost)]->(k:Three) return count(p);
+            """
+        Then the result should be:
+            | count(p) |
+            | 1 |
```

Comment:
This is not a good test since we don't know what's the cost of the shortest path